pkgs/top-level/all-packages.nix

```diff
@@ -11433,7 +11420,7 @@ let
 
   eiskaltdcpp = callPackage ../applications/networking/p2p/eiskaltdcpp { lua5 = lua5_1; };
 
-  qemu = callPackage ../applications/virtualization/qemu { };
+  qemu = callPackage ../applications/virtualization/qemu (config.qemu or {});
```

Comment:
Okay. Let's talk about this. For me this "there are packageOverrides" sounds like. "Why would you need NixOS when you could slap this bunch of packages together and then configure all of them by hand". I can't see why having a readable mechanism for setting default values is a bad thing.

If what you are actually want to say is that this *looks* ugly, then okay, a valid point. But that is trivially fixed by adding a version of `callPackage` that does this stuff automatically (this patch is in queue). After which I'd say that it is `packageOverrides` that look ugly.

Finally, I would like to get NixOS-like configuration for packages' flags. Because currently with package overrides it's very easy to get several versions of the same package when, in fact, it would be sufficient to have only one. Do `egrep -rn '\.override' nixos/modules`, some of these will actually compile another version when all they need is a flag to be turned on.

To solve this one could make NixOS override packages globally, but this will eventually result in a confusion when user overrides a package and then NixOS overrides it too and who knows what will the options actually be set to.

Note, I'm not saying that overriding doesn't have it uses, it's most certainly does (and some of the uses in `nixos/modules` are pretty valid). I'm saying that you probably don't want NixOS modules to override packages, but I can see when and why NixOS modules may want to change the defaults (see, e.g. #6969).

That was an expansion of https://github.com/NixOS/nixpkgs/pull/6968#issuecomment-85718570 a), there is also https://github.com/NixOS/nixpkgs/pull/6968#issuecomment-85718570 b) which is about setting common options (a patch for which is, again, in a queue) and, finally, https://github.com/NixOS/nixpkgs/pull/6968#issuecomment-85721122